carbon-design-system/carbon-addons-iot-react
# feat(hierarchy-list): indeterminate checkbox handling to items in MultipleNesting mode.

Closes #3004 

### Depends on #3064 and #3065

**Summary**

- Add indeterminate state handling to HierarchyList in MultipleNesting mode

**Change List (commits, features, bugs, etc)**

- add indeterminateIds state and pass that to the list
- adjust the `handleEditModeSelect` function to mark the parent as selected if all children are.
- add useEffect to HierarchyList to keep indeterminateIds updated as selected ids are changed.
- add tests to confirm

**Acceptance Test (how to verify the PR)**

1. https://deploy-preview-3066--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-hierarchylist--with-nested-reorder
2. turn on MultipleNesting mode
3. confirm New York Mets is indeterminate
4. Check the New York Mets checkbox
5. confirm New York Mets is still indeterminate (b/c of the locked row)
6. Check the Atlanta Braves
7. Uncheck an individual player
8. Confirm the Atlanta Braves is indeterminate
9. Re-check the player
10. 8. Confirm the Atlanta Braves is checked
11. repeat with isVirtualList

**Regression Test (how to make sure this PR doesn't break old functionality)**

- tests here



**Things to look for during review**
